pycqed/measurement/waveform_control/pulse.py

```diff
@@ -131,3 +131,10 @@ def chan_wf(self, channel, tvals):
         """
         raise NotImplementedError('chan_wf() not implemented for {}'
                                   .format(str(type(self))[1:-1]))
+
+    @staticmethod
+    def pulse_params():
+        """
+        Returns a dictionary of pulse parameters and initial values.
+        """
+        raise NotADirectoryError('pulse_params() not implemented for your pulse')
```

Comment:
why use this exception? I guess NotImplementedError was meant here.

Reply:
Oops, yes auto-complete set me a trap ;-)